### ISSUE TITLE
Enhancement: TigerVNC x64 Bit

### ISSUE THREAD

**Issue by gsmurray** [TigerVNC/tigervnc#695]
@bphinz 

MAC OS Mojave will not support x32 bit applications.

Under High-Sierra the following Message is received when Starting an x32 Bit App:

"App" is not optimized for you MAC
“This app needs to be updated by its developer to improve compatibility”

This is actually a warning that High-Sierra is the last MAC OS that will support x32 bit Applications.

Please Provide an x64 Bit Version of the Tiger VNC Application for MAC OS

Thanks,

Scott

**Comment by bphinz** [comment]
The latest nightly build is x64.  Please give it a try.

**Comment by gsmurray** [comment]
Excellent, Thanks Brian,

I’ve installed and run, looks good thus far.
Will do some functional testing today and report any issues.

Also passed this to our development team to test.


Scott Murray | 613-670-5388 | 613-793-5792

**Comment by gsmurray** [comment]
Received a Response from our Testing Team; I also Experienced this issue.

During Initial Startup, we received the standard Unidentified App Dialog from the Gatekeeper.
Seleting the “Open” button should have Opened the App and Exempted it from the Gatekeeper Policy.

Our Result was that the App Did not start after selecting the “Open” Button.
Subsequent Starts were successful, however.

Browsing logs did not provide any hints.

Functional Testing was successful; no issues reported with Copy/Paste, Video, Re-sizing, Authentication, etc.

Thanks,


Scott Murray | 613-670-5388 | 613-793-5792

**Comment by bphinz** [comment]
I saw this too, not sure what to make of it.  Nothing changed with any part of the build except the compile flags.

**Comment by matlupi** [comment]
> The latest nightly build is x64. Please give it a try.

Where can I find it?

**Comment by CendioOssman** [comment]
From the nightly build link on our homepage.

**Comment by CendioOssman** [comment]
@bphinz, can we close this issue now?

**Comment by gsmurray** [comment]
Sorry Brian for not getting back to you sooner. 
Thanks for actioning this.

**Comment by jwcto** [comment]
Got the same warning when running Mojave. Now I'm on Catalina and Tiger VNC Viewer 1.9.0 will not launch and shows the message "Tiger VNC Viewer 1.9.0.app needs to be updated"

**Comment by gsmurray** [comment]
As I reported in my testing I did receive a message on the First start of the Application.
Subsequent attempt to open the Application did not provide the message.

Are you continually receiving the messages on Catalina?

Thanks,

Scott

**Comment by jwcto** [comment]
Yes. I even restarted my laptop a couple of times. And downloaded a fresh copy of Tiger VNC. Same result.

John Phillip Whittemore
Chairman & CEO
Compliance Training Online

+1 (512) 410-7335 x1

jw@compliancetrainingonline.com
www.compliancetrainingonline.com

https://www.facebook.com/compliancetrainingonline/
https://www.linkedin.com/company/compliance-training-online
https://twitter.com/Compliance_CTO
https://compliancetrainingonline.tumblr.com

**Comment by gsmurray** [comment]
Thanks John,

I had the same result, even with the latest beta version.
I suspect this is due to the universal binary that is used for the Intel-based Mac.
The x32 bit code within is preventing the Application from Starting.

I will need this Application in my environment so I will open an issue with the Developers




George Scott Murray
O | 6136705388
M | 6137935792

**Comment by gsmurray** [comment]
John,

I was able to get V1.9.80 to open after multiple Attempts.
Once opened, subsequent starts of the Application are successful.

Still opening a bug for this, Just though you should know.


George Scott Murray
O | 6136705388
M | 6137935792

**Comment by jwcto** [comment]
Thanks. Yea, it still doesn’t work for me. Apparently Catalina won’t allow applications it sees as 32bit to launch at all.

John Phillip Whittemore
Chairman & CEO
Compliance Training Online

+1 (512) 410-7335 x1

jw@compliancetrainingonline.com
www.compliancetrainingonline.com

https://www.facebook.com/compliancetrainingonline/
https://www.linkedin.com/company/compliance-training-online
https://twitter.com/Compliance_CTO
https://compliancetrainingonline.tumblr.com

**Comment by gsmurray** [comment]
Apple was stating this for a while; Mojave was to be the last OS that would support x32.
That is why I open the original Enhancement Request.

I’ve now opened TigerVNC 1.9.80 failing to start on MAC OS Catalina #881

Hopefully this will be addressed soon.


George Scott Murray
O | 6136705388
M | 6137935792

**Comment by gsmurray** [comment]
I did find a work around to get the Application opening; Seems it is being blocked by Apple security
as TigerVNC are not an Identified publisher at Apple, Not because it contains x32 code:



  1.  Download 1.9.80 from http://tigervnc.bphinz.com/nightly/
  2.  Click on .dmg ->copy the app in your Applications folder;
  3.  Double click shows you "Apple cannot check it for malicious software"->Click OK
  4.  Go to System Preferences > Security & Privacy > General (Tab) > clicked on “Open Anyway” nearby TigerVPN app;
  5.  This will Show the show you "Apple cannot check it for malicious software" dialog again->Select OK
  6.  Attempt to open the Application again by Double-clicking
Next time you are going to try to open the app from applications, you will be able to do it just with a double click as any other app.


George Scott Murray
O | 6136705388
M | 6137935792

**Comment by jwcto** [comment]
“Open Anyway” is gone from Catilina.

John Phillip Whittemore
Chairman & CEO
Compliance Training Online

+1 (512) 410-7335 x1

jw@compliancetrainingonline.com
www.compliancetrainingonline.com

https://www.facebook.com/compliancetrainingonline/
https://www.linkedin.com/company/compliance-training-online
https://twitter.com/Compliance_CTO
https://compliancetrainingonline.tumblr.com
